exercises/binary-search/src/test/java/BinarySearchTest.java

```diff
@@ -1,18 +1,23 @@
 
 import org.junit.Ignore;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.NoSuchElementException;
 
 import static org.junit.Assert.assertEquals;
 
 public class BinarySearchTest {
-
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+    
     @Test
-    public void findsAValueInAnArrayWithOneElement() {
+    public void findsAValueInAnArrayWithOneElement() throws NoSuchElementException {
```

Comment:
So going according to the bottom-line of the article, we should be making this a checked Exception. To do so, I can make a custom Exception and throw it like I was doing earlier. Any suggestions on the name of the Exception? I had used "ValueNotInArrayException" but does it make it too specific? Or should we keep a more generic name and throw the specific part of it as the exception message?

Reply:
I think it's good to be specific since we're making it specifically for this exercise. So something like `ValueNotInArrayException` or maybe `ValueNotFoundException` would be fine 🙂 

Also just so you know, when adding a custom exception we usually add the implementation to the starter implementation as well, see [the bank account exercise](https://github.com/exercism/java/tree/master/exercises/bank-account) for an example 🙂